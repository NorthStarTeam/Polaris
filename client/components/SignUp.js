import React, { useContext } from 'react';
import { Redirect } from 'react-router-dom';
import { UserContext } from './AccState';

const SignUp = () => {
  const userObj = useContext(UserContext);
  const { username, setName, password, setPass, isLogin, setLogin } = userObj;

  const handleInputName = event => {
    setName(event.target.value);
  };
  const handleInputPass = event => {
    setPass(event.target.value);
  };

  const handleSubmit = e => {
    e.preventDefault();
    fetch('/signup', {
      headers: { 'Content-type': 'application/json' },
      method: 'POST',
      body: JSON.stringify({ username, password }),
    })
      .then(res => res.json())
      .then(data => {
        console.log('signup from server =', data);
        if (data.exist) {
          setLogin(false);
          setName('');
          setPass('');
        } else setLogin(true);
      });
  };

  if (isLogin)
    return (
      <Redirect
        to={{
          pathname: '/dashboard',
        }}
      />
    );

  return (
    <div className="div-signup">
      <h1>Sign Up</h1>
      <div className="signup-box">
        <form onSubmit={e => handleSubmit(e)} className="signup-form">
          <input
            className="signup-input"
            placeholder="Username"
            type="text"
            value={username}
            onChange={handleInputName}
          />
          <input
            className="signup-input"
            placeholder="Password"
            type="text"
            value={password}
            onChange={handleInputPass}
          />
          <input className="button" type="submit" value="Signup"></input>
        </form>
      </div>
    </div>
  );
};

export default SignUp;
