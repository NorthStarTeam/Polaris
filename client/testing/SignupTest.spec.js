import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import SignUp from '../components/SignUp';

describe('SignUp', () => {
    
    //snapshot test
    test('SignUp renders correctly', () => {
        const component = renderer.create(<MemoryRouter> <SignUp /> </MemoryRouter>)
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot();
    })

    //enzyme tests
    it('renders a div with the class of signup-box', () => {
        const wrapper = mount(<MemoryRouter> <SignUp /> </MemoryRouter>)
        expect(wrapper.find('div.signup-box').length).toEqual(1)
    })


})