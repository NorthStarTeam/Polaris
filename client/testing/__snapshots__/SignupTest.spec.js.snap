// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp SignUp renders correctly 1`] = `
Array [
  " ",
  <div>
    <h1>
      Sign Up
    </h1>
    <div
      className="signup-box"
    >
      <form
        className="signup-form"
      >
        <input
          className="userName"
          name="username"
          placeholder="Username"
        />
        <input
          className="password"
          name="password"
          placeholder="Password"
        />
        <input
          className="signup-button"
          type="submit"
          value="Signup"
        />
      </form>
    </div>
  </div>,
  " ",
]
`;
